examples/run_generic_echo.py

```diff
@@ -0,0 +1,121 @@
+phase_cycling = True
+if phase_cycling:
+    ph1_cyc = r_[0,1,2,3]
+    ph2_cyc = r_[0]
+    nPhaseSteps = len(ph1_cyc)
+if not phase_cycling:
+    nPhaseSteps = 1
+# }}}    
+prog_p90_us = prog_plen(config_dict['p90_us'])
+prog_p180_us = prog_plen(2*config_dict['p90_us'])
+# {{{check total points
+total_pts = nPoints * nPhaseSteps# * config_dict['nEchoes']
+assert total_pts < 2**14, "You are trying to acquire %d points (too many points) -- either change SW or acq time so nPoints x nPhaseSteps is less than 16384"%total_pts
+# }}}
+# {{{basic phasecycling
+data = generic(
+    ppg_list=[
+        ("phase_reset", 1),
+        ("delay_TTL", config_dict["deblank_us"]),
+        ("pulse_TTL", prog_p90_us, "ph1", ph1_cyc),
+        ("delay", config_dict["tau_us"]),
+        ("delay_TTL", config_dict["deblank_us"]),
+        ("pulse_TTL", prog_p180_us, "ph2", ph2_cyc),
+        ("delay", config_dict["deadtime_us"]),
+        ("acquire", config_dict["acq_time_ms"]),
+        ("delay", config_dict["repetition_us"]),
+    ],
+    nScans=config_dict["nScans"],
+    indirect_idx=0,
+    indirect_len=config_dict["nEchoes"],
+    adcOffset=config_dict["adc_offset"],
+    carrierFreq_MHz=config_dict["carrierFreq_MHz"],
+    nPoints=nPoints,
+    acq_time_ms=config_dict["acq_time_ms"],
+    SW_kHz=config_dict["SW_kHz"],
+    ret_data=None,
+)
+## {{{ chunk and save data
+data.set_prop('postproc_type','proc_Hahn_echoph')
+data.set_prop("acq_params", config_dict.asdict())
+data.name(config_dict["type"] + "_" + str(config_dict["cpmg_counter"]))
+data.chunk(
+    "t", 
+    ["ph1", "t2"], 
+    [4,-1])
+data.setaxis('ph1',r_[0.,1.,2.,3.]/4)
+# }}}
+target_directory = getDATADIR(exp_type="ODNP_NMR_comp/Echoes")
+filename_out = filename + ".h5"
+nodename = data.name()
+if os.path.exists(f"{filename_out}"):
+    print("this file already exists so we will add a node to it!")
+    with h5py.File(
+        os.path.normpath(os.path.join(target_directory, f"{filename_out}"))
+    ) as fp:
+        if nodename in fp.keys():
+            print("this nodename already exists, so I will call it temp_cpmg")
+            data.name("temp_cpmg")
+            nodename = "temp_cpmg"
+    data.hdf5_write(f"{filename_out}", directory=target_directory)
+else:
+    try:
+        data.hdf5_write(f"{filename_out}", directory=target_directory)
+    except:
+        print(
+            f"I had problems writing to the correct file {filename}.h5, so I'm going to try to save your file to temp_cpmg.h5 in the current h5 file"
+        )
+        if os.path.exists("temp_cpmg.h5"):
+            print("there is a temp_cpmg.h5 already! -- I'm removing it")
+            os.remove("temp_cpmg.h5")
+            data.hdf5_write("temp_cpmg.h5")
+            print(
+                "if I got this far, that probably worked -- be sure to move/rename temp_cpmg.h5 to the correct name!!"
+            )
+print("\n*** FILE SAVED IN TARGET DIRECTORY ***\n")
+print(("Name of saved data", data.name()))
```

Comment:
I might be missing it, but for every pulse program, you want to be sure that you're writing the variable that it reads to determine the postprocessing type --> I think that it's "postproc_type". And when you modify the program, you should increase the version number.

Reply:
so, for these pulse programs, I would actually say that *we should not be visualizing the data at all*

*However*, you should add an example to proc scripts that's something like this:
```
"""
Show data with postproc
====================
`py proc_raw_data.py NODENAME FILENAME EXP_TYPE`

Fourier transforms (and any needed data corrections for older data) are performed according to the `postproc_type` attribute of the data node.
This script plots the result, as well as signal that's averaged along the `nScans` dimension.
"""
from pyspecProcScripts.load_data import lookup_table
import sys
import os
assert len(sys.argv) == 4
d = find_file(sys.argv[2], exp_type=sys.argv[3], expno=sys.argv[1],
                      lookup=lookup_table)
with figlist_var() as fl:
    d.squeeze()
    fl.next("raw data")
    def image_or_plot(d):
        if len(d.dimlabels) > 2:
            fl.image(d)
        else:
            fl.plot(d)
    image_or_plot(d)
    if 'nScans' in d.dimlabels:
        d.mean('nScans')
        fl.next("signal averaged along nScans")
        image_or_plot(d)
 ```
 then you can simply be sure that the ppgs print NODENAME FILENAME EXP_TYPE at the end, so you can copy and paste to run this.  (or even print out the command you want to run to show the data)